stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2513,6 +2805,47 @@ internal struct _Native${Self}Storage<${TypeParameters}> {
   /// We store a RawBuffer so that this can be an empty singleton
   internal var _buffer: RawBuffer
 
+  /// Creates a Storage with a buffer that is typed, but doesn't understand
+  /// Hashing. Mostly for bridging; prefer `create(capacity:)`.
```

Comment:
It's `create(capacity:)` that's available when `K: !Hashable`. It must be a different ctor because we instantiate a different class as the buffer.